<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>683</width>
    <height>422</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TCP/UDP port listener</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tabWelcome">
       <attribute name="title">
        <string>Welcome</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QGroupBox" name="boxListeners">
            <property name="title">
             <string>Listeners</string>
            </property>
            <layout class="QFormLayout" name="formLayout">
             <item row="0" column="0" colspan="2">
              <widget class="QLabel" name="lblAbout">
               <property name="text">
                <string>To start listening any TCP/UDP port or Local Socket (pipe) press:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="btnTcp">
               <property name="text">
                <string>TCP port</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="lblTcp">
               <property name="text">
                <string>via QTcpServer</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QPushButton" name="btnUdp">
               <property name="text">
                <string>UDP port</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLabel" name="lblUdp">
               <property name="text">
                <string>via QUdpSocket</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QPushButton" name="btnSocket">
               <property name="text">
                <string>Socket</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLabel" name="lblSocket">
               <property name="text">
                <string>via QLockalSocket</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Modbus-TCP</string>
            </property>
            <layout class="QFormLayout" name="formLayout_2">
             <item row="0" column="0" colspan="2">
              <widget class="QLabel" name="lblAboutModbus">
               <property name="text">
                <string>To deal with Modbus TCP press:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="btnModbusTcpServer">
               <property name="text">
                <string>Modbus TCP Server</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>via QModbusTcpServer</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QPushButton" name="btnModbusTcpClient">
               <property name="text">
                <string>Modbus TCP Client</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>via QModbusTcpClient</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>116</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>683</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuConnect">
    <property name="title">
     <string>&amp;Listen</string>
    </property>
    <addaction name="actionTCP_port"/>
    <addaction name="actionUDP_port"/>
    <addaction name="actionSocket"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuModbus">
    <property name="title">
     <string>Modbus</string>
    </property>
    <addaction name="actionModbusTcpServer"/>
    <addaction name="actionModbusTcpClient"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionDataSender"/>
   </widget>
   <addaction name="menuConnect"/>
   <addaction name="menuModbus"/>
   <addaction name="menuTools"/>
  </widget>
  <action name="actionTCP_port">
   <property name="text">
    <string>&amp;TCP port</string>
   </property>
  </action>
  <action name="actionUDP_port">
   <property name="text">
    <string>&amp;UDP port</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="actionSocket">
   <property name="text">
    <string>&amp;Socket</string>
   </property>
  </action>
  <action name="actionModbusTcpServer">
   <property name="text">
    <string>Modbus TCP Server</string>
   </property>
  </action>
  <action name="actionModbusTcpClient">
   <property name="text">
    <string>Modbus TCP Client</string>
   </property>
  </action>
  <action name="actionDataSender">
   <property name="text">
    <string>Data Sender</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
